### PR TITLE
[MRG] Remove acq_optimizer='auto' and set default to 'sampling'

### PR DESCRIPTION
**Pull request by MechCoder**
Fixes https://github.com/scikit-optimize/scikit-optimize/issues/274

See benchmarks:

1. https://github.com/scikit-optimize/scikit-optimize/pull/262#issuecomment-269493170
2. https://github.com/scikit-optimize/scikit-optimize/pull/262#issuecomment-260835534

### CODE REVIEW

**Comment by glouppe** [comment]
What shall we do with this now?

**Comment by MechCoder** [comment]
There is definitely some issues with how we handle Integer dimensions. I'll get back to this as soon as I have some solid reasoning behind that.

**Comment by MechCoder** [comment]
(In the context of switching the default from "auto" to "lbfgs" according to the original issue)

**Comment by MechCoder** [comment]
Running it on https://scikit-optimize.github.io/notebooks/hyperparameter-optimization.html there does not seem to be a huge difference. I think we can remove it.

Without K-best:

| Acquisition Function | Mean Minimum | Std |
| --- | --- | --- |
| lbfgs | 2.884 | 0.037 |
| sampling | 2.844   | 0.024 |

With K-best

| Acquisition Function | Mean Minimum | Std |
| --- | --- | --- |
| lbfgs | 2.925 | 0.032 |
| sampling | 2.917   | 0.035 |

**Comment by glouppe** [comment]
What do you call "K-best"?

**Comment by MechCoder** [comment]
I had meant with and without the preprocessing using SelectKBest

**Comment by MechCoder** [comment]
Another observation, I have to increase `n_restarts_optimizer` further to give a non-constant predictions for estimating piecewise constant functions.

```python
import numpy as np
import matplotlib.pyplot as plt

from skopt.learning import GaussianProcessRegressor
from skopt.learning.gaussian_process.kernels import ConstantKernel
from skopt.learning.gaussian_process.kernels import Matern

def f(x):
    return int(x)**2 - 1

x = np.linspace(0, 20, 61)[:60]
y = [f(xi) for xi in x]

for startptr, endptr in zip(range(0, 61, 3), range(3, 62, 3)):
    plt.plot(x[startptr: endptr], y[startptr: endptr], "black")

cov_amplitude = ConstantKernel(1.0, (0.01, 1000.0))
matern = Matern(length_scale=1.0, length_scale_bounds=[0.01, 100], nu=2.5)

gp = GaussianProcessRegressor(
    kernel=cov_amplitude * matern, normalize_y=True, random_state=0,
    alpha=0.0, n_restarts_optimizer=5)
gp.fit(np.reshape(x, (-1, 1)), y)

x_unseen = np.linspace(0, 20, 201)[:200]
y_pred = gp.predict(np.reshape(x_unseen, (-1, 1)))

for startptr, endptr in zip(range(0, 201, 20), range(20, 221, 20)):
    plt.plot(x_unseen[startptr: endptr], y_pred[startptr: endptr], "red")
plt.show()
```

With `n_restarts_optimizer=5`, the graph looks like



With `n_restarts_optimizer=2`, the graph looks like

**Comment by glouppe** [comment]
Interesting! This is also what we observe in the BO notebook, everything is flat except some spikes, just like your second plot. Maybe there is something wrong in the training procedure of the GP that we should mention upstream in scikit-learn? In the meantime, shall we increase `n_restarts_optimizer` further?

File: skopt/optimizer/gp.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def gp_minimize(func, dimensions, base_estimator=None,
         - `"EI"` for negative expected improvement.
         - `"PI"` for negative probability of improvement.
 
-    * `acq_optimizer` [string, `"auto"`, `"sampling"` or `"lbfgs"`, default=`"auto"`]:
+    * `acq_optimizer` [string, `"sampling"` or `"lbfgs"`, default=`"sampling"`]:
```
Comment (glouppe) [inline skopt/optimizer/gp.py]: Dont you think we should set instead LBFGS as default?

**Comment by MechCoder** [comment]
> Maybe there is something wrong in the training procedure of the GP that we should mention upstream in scikit-learn? 

When I had a look sometime before, the implementation seemed to be correct. I just think it gets stuck in a local minima. I can have a closer look though.

**Comment by MechCoder** [comment]
Btw, I closed the pull request because I messed up some git stuff

**Pull request closed without merging**